internal/io/csv.go

```diff
@@ -98,6 +101,40 @@ func ReadCSV(reader io.Reader, conf CSVConfig) (map[string]interface{}, []string
 		}
 	}
 
+	if conf.MissingColumnNameAlias != "" {
+		for i, name := range headers {
+			if name == "" {
+				headers[i] = conf.MissingColumnNameAlias
+			}
+		}
+	}
+
+	if conf.AddColumnIndexToDuplicateColumns {
+		headersMap := make(map[string]int)
+		// Loop through all columsn and find the duplicates.
+		for i, h := range headers {
+			headersMap[h] = i
```

Comment:
I personally would prefer if this was the first index of the header rather than the last.
Eg. `COL,COL,COL` -> `COL,COL0,COL1`, right now it's `COL,COL,COL` -> `COL0,COL1,COL`.